compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -746,22 +746,40 @@ fn encode_ty<'tcx>(
 }
 
 /// Transforms predicates for being encoded and used in the substitution dictionary.
-fn transform_predicates<'tcx>(
+fn transform_dyn_predicates<'tcx>(
     tcx: TyCtxt<'tcx>,
     predicates: &List<ty::PolyExistentialPredicate<'tcx>>,
+    parents: &mut Vec<Ty<'tcx>>,
+    options: TransformTyOptions,
 ) -> &'tcx List<ty::PolyExistentialPredicate<'tcx>> {
-    tcx.mk_poly_existential_predicates_from_iter(predicates.iter().filter_map(|predicate| {
-        match predicate.skip_binder() {
-            ty::ExistentialPredicate::Trait(trait_ref) => {
-                let trait_ref = ty::TraitRef::identity(tcx, trait_ref.def_id);
-                Some(ty::Binder::dummy(ty::ExistentialPredicate::Trait(
-                    ty::ExistentialTraitRef::erase_self_ty(tcx, trait_ref),
-                )))
-            }
-            ty::ExistentialPredicate::Projection(..) => None,
-            ty::ExistentialPredicate::AutoTrait(..) => Some(predicate),
-        }
-    }))
+    let mut predicates: Vec<_> = predicates
+        .iter()
+        .map(|predicate| {
+            predicate.map_bound(|predicate| match predicate {
+                ty::ExistentialPredicate::Trait(ty::ExistentialTraitRef { def_id, args }) => {
+                    ty::ExistentialPredicate::Trait(ty::ExistentialTraitRef {
+                        def_id,
+                        args: transform_args(tcx, args, parents, options),
+                    })
+                }
+                ty::ExistentialPredicate::Projection(ty::ExistentialProjection {
+                    def_id,
+                    args,
+                    term,
+                }) => ty::ExistentialPredicate::Projection(ty::ExistentialProjection {
+                    def_id,
+                    args: transform_args(tcx, args, parents, options),
+                    term: match term.unpack() {
+                        TermKind::Ty(ty) => transform_ty(tcx, ty, parents, options).into(),
+                        TermKind::Const(ct) => ct.into(),
+                    },
+                }),
+                ty::ExistentialPredicate::AutoTrait(..) => predicate,
+            })
+        })
+        .collect();
+    predicates.sort_by(|a, b| a.skip_binder().stable_cmp(tcx, &b.skip_binder()));
```

Comment:
why is this needed? please add a comment why we have to sort here

Reply:
Actually, this doesn't matter. We only sort by def path rn lol